tools/code_policy_check.py

```
@@ -0,0 +1,262 @@
+        normalized = normalized.lstrip("\ufeff")
+    return normalized
+
+
+def _has_spdx_header(lines):
+    for line in lines[:5]:
+        if SPDX_IDENTIFIER in line:
+            return True
+    return False
+
+
+def _check_headers(path, lines, context):
```

Ruff (0.14.6)

Missing return type annotation for private function `_check_headers`

Add return type annotation: `str | None`

(ANN202)